server/src/main/java/org/elasticsearch/rest/action/document/RestUpdateAction.java

```diff
@@ -50,9 +57,16 @@ public String getName() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
-        UpdateRequest updateRequest = new UpdateRequest(request.param("index"),
-            request.param("type"),
-            request.param("id"));
+        UpdateRequest updateRequest;
+        if (request.hasParam("type")) {
+            deprecationLogger.deprecated(TYPES_DEPRECATION_MESSAGE);
```

Comment:
Do we want to use `deprecatedAndMaybeLog` instead of `deprecated` not to log same deprecation warnings many times?

Reply:
Thanks a lot, I hadn't seen this method. I think we should use it more generally for types deprecation -- I will go over and update some of our other logging statements as well.